src/pages/TransactionsPage/TransactionsForm/TransactionsForm.js

```diff
@@ -0,0 +1,180 @@
+import { StyledTransactionForm } from "./TransactionsForm.styled";
+import { DatePicker } from "../../../components/form/DatePicker/DatePicker";
+import { Input } from "../../../components/form/Input/Input";
+import { RadioButton } from "../../../components/form/RadioButton/RadioButton";
+import { Dropdown } from "../../../components/form/Dropdown/Dropdown";
+import { useInputState } from "../../../services/hooks/useInputState";
+import { useState, useEffect } from "react";
+import { validate } from "../../../services/helpers/formValidationRules";
+import Modal from "../../../components/Modal";
+import { useModal } from "../../../services/hooks/useModal";
+
+export const TransactionForm = () => {
+  const { isVisible, toggleVisibility } = useModal();
+
+  const [title, setTitle, resetTitle] = useInputState("");
+  const [description, setDescription, resetDescription] = useInputState("");
+  const [amount, setAmount, resetAmount] = useInputState(0);
+  const [type, setType, resetType] = useInputState("income");
+  const [date, setDate, resetDate] = useInputState("");
+  const [category, setCategory] = useState("select");
+  const [paymentType, setPaymentType, resetPaymentType] = useInputState("cash");
+  const [formErrors, setFormErrors] = useState({});
+  const [isTouched, setIsTouched] = useState({
+    title: false,
+    description: false,
+    amount: false,
+    date: false,
+    category: false,
+  });
+
+  const validateForm = () => {
+    const formValues = {
+      title: title,
+      description: description,
+      amount: amount,
+      date: date,
+      category: category,
+    };
+    const errors = validate(formValues);
+    setFormErrors(errors);
+  };
+
+  const handleBlur = (field) => {
+    setIsTouched((values) => ({ ...values, [field]: true }));
+  };
+
+  useEffect(() => {
+    validateForm();
+  }, [title, description, amount, date, category]);
+
+  const clearFormValues = () => {
+    resetTitle();
+    resetDescription();
+    resetAmount();
+    resetType();
+    resetDate();
+    resetPaymentType();
+    setCategory("select");
+    setIsTouched({
+      title: false,
+      description: false,
+      amount: false,
+      date: false,
+      category: false,
+    });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setIsTouched({
+      title: true,
+      description: true,
+      amount: true,
+      date: true,
+      category: true,
+    });
+    validateForm();
```

Comment:
i think this is reduntant, because this func `validateForm` is called in useEffect on each form element change